_pytest/compat.py

```diff
@@ -237,5 +237,7 @@ def safe_str(v):
         try:
             return str(v)
         except UnicodeError:
+            if not isinstance(v, unicode):
+                v = unicode(v)
             errors = 'replace'
             return v.encode('ascii', errors)
```

Comment:
Sounds like a bad idea - things like the Window console will throw exceptions when trying to print UTF-8 with Python < 3.6.